src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py

```diff
@@ -781,12 +808,14 @@ def __call__(
 
         # 5. Prepare timesteps
         self.scheduler.set_timesteps(num_inference_steps, device=device)
-        timesteps, num_inference_steps = self.get_timesteps(num_inference_steps, strength, device)
+        timesteps, num_inference_steps = self.get_timesteps(num_inference_steps, first_inference_step, strength, device)
         latent_timestep = timesteps[:1].repeat(batch_size * num_images_per_prompt)
-
+        add_noise = True
+        if first_inference_step is not None:
+            add_noise = False
```

Comment:
```suggestion
        add_noise = True
        if denoising_start > 0.0:
            add_noise = False
```

I like this mechanism for now! Nice

Reply:
I have altered this to take an explicitly-defined parameter, so that noise may optionally be added for more creative/variable outputs while retaining the ability to properly align timesteps from a previous instance.